hazelcast-hibernate/hazelcast-hibernate4/src/main/java/com/hazelcast/hibernate/local/LocalRegionCache.java

```diff
@@ -99,6 +100,7 @@ public LocalRegionCache(final String name, final HazelcastInstance hazelcastInst
         } else {
             topic = null;
         }
+        hazelcastInstance.getCluster().addMembershipListener(this);
```

Comment:
In addition (if anyone's curious) there is also a `CleanupService` that holds a strong reference to almost all the `LocalRegionCache`s ever constructed and calls `cleanup()` on them every 60 seconds, forever.  So even if they went out of scope they will not be GC'd.